crypto/x509/x509_test.cc

```diff
@@ -2301,6 +2302,32 @@ TEST(X509Test, Ed25519Sign) {
   ASSERT_TRUE(SignatureRoundTrips(md_ctx.get(), pub.get()));
 }
 
+TEST(X509Test, DilithiumSignVerifyCert) {
+  //generate the dilithium key
+  EVP_PKEY_CTX *dilithium_pkey_ctx = EVP_PKEY_CTX_new_id(EVP_PKEY_DILITHIUM3, nullptr);
+  ASSERT_NE(dilithium_pkey_ctx, nullptr);
+  EVP_PKEY *dilithium_pkey = EVP_PKEY_new();
+  ASSERT_NE(dilithium_pkey, nullptr);
+  EXPECT_TRUE(EVP_PKEY_keygen_init(dilithium_pkey_ctx));
+  EXPECT_TRUE(EVP_PKEY_keygen(dilithium_pkey_ctx, &dilithium_pkey));
+
+  //generate the cert
+  bssl::UniquePtr<X509> leaf =
+      MakeTestCert("Intermediate", "Leaf", dilithium_pkey, /*is_ca=*/false);
+  ASSERT_TRUE(leaf);
+
+  //sign the cert
+  bssl::ScopedEVP_MD_CTX md_ctx;
+  EVP_DigestSignInit(md_ctx.get(), nullptr, nullptr, nullptr, dilithium_pkey);
+  ASSERT_TRUE(X509_sign_ctx(leaf.get(), md_ctx.get()));
+
+  //verify the cert
+  ASSERT_TRUE(X509_verify(leaf.get(), dilithium_pkey));
+
+  EVP_PKEY_CTX_free(dilithium_pkey_ctx);
+  EVP_PKEY_free(dilithium_pkey);
```

Comment:
can we add tests for failure modes? for example, change something in the cert to make it invalid and then check that the verification indeed fails?

Reply:
added in [19b192e](https://github.com/awslabs/aws-lc/pull/759/commits/19b192eb20625b1e955218ee799371bc57a898d4). We now test the extraction of the public key from a CertFromPEM, as well as verification, and we also test that verification fails on a cert with an explicit `NULL` in the signature algorithm - just as is done for x25519.